variables.tf

```diff
@@ -9,3 +13,257 @@ variable "tags" {
+  default     = []
+  description = "SGs permitted access to RDS"
+  type        = list(string)
+}
+
+variable "db_allowed_access_cidrs" {
+  default     = []
+  description = "CIDRs permitted access to RDS"
+  type        = list(string)
+}
+
+variable "db_engine_version" {
+  default     = "11"
+  description = "engine version to run"
+  type        = string
+}
+
+variable "db_storage_size" {
+  description = "Size in DB (in GB)"
+  type        = number
+}
+
+variable "db_instance_class" {
+  description = "DB Instance Size"
+  type        = string
+}
+
+variable "db_multi_az" {
+  default     = false
+  description = "If true, DB will be configured in multi-AZ mode"
+  type        = bool
+}
+
+variable "db_monitoring_role_arn" {
+  default     = null
+  description = "ARN for Database Monitoring (required for performance insights)"
+  type        = string
+}
+
+variable "db_monitoring_interval" {
+  default     = 0
+  description = "Enhanced monitoring interval (5-60 seconds, 0 to disable)"
+  type        = number
+}
+
+variable "db_parameters" {
+  description = "DB parameters (downstream module defaults will be used if not specified)"
+
+  default = [
+    {
+      name  = "client_encoding"
+      value = "UTF8"
+    }
+  ]
+
+  type = list(object({
+    name  = string
+    value = string
+  }))
+}
+
+variable "db_performance_insights_enabled" {
+  default     = false
+  description = "Whether or not to enable DB performance insights"
+  type        = bool
+}
+
+variable "db_subnet_group" {
+  description = "Database subnet group"
+  type        = string
+}
+
+variable "db_vpc_id" {
+  default     = null
+  description = "VPC ID for database (if omitted, the value for `vpc_id` is used instead)"
+  type        = string
+}
+
+########################################
+# Networking Vars
+########################################
+
+variable "availability_zone" {
+  description = "Specify the availability zone that the instance will be deployed in. Because Bitbucket requires an EBS volume for data and can't use EFS, the value of `availability_zone` must match the AZ associated with the value of `asg_subnet`."
+  type        = string
+}
+
+variable "elb_additional_sg_tags" {
+  default     = {}
+  description = "Additional tags to apply to the ELB security group. Useful if you use an external process to manage ingress rules."
+  type        = map(string)
+}
+
+variable "elb_allowed_cidr_blocks" {
+  default     = ["0.0.0.0/0"]
+  description = "List of allowed CIDR blocks. If `[]` is specified, no inbound ingress rules will be created"
+  type        = list(string)
+}
+
+variable "elb_certificate" {
+  description = "ARN of certificate to associate with ELB"
+  type        = string
+}
+
+variable "elb_internal" {
+  default     = true
+  description = "Create as an internal or internet-facing ELB"
+  type        = bool
+}
+
+variable "elb_subnets" {
+  description = "Subnets to associate ELB to"
+  type        = list(string)
+}
```

Comment:
Does the ELB really need to be associated with multiple subnets if we're forcing the instance to always be in the same subnet?

Reply:
It doesn't _need_ to be, but an ELB defaults to cross AZ load balancing, so I could either forcibly make that false to use a single subnet or allow the user the choice.